runtime/src/main/java/org/corfudb/runtime/view/stream/SingleStreamView.java

```diff
@@ -0,0 +1,291 @@
+package org.corfudb.runtime.view.stream;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.UUID;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+
+import javax.annotation.Nonnull;
+import lombok.extern.slf4j.Slf4j;
+
+import org.corfudb.protocols.wireprotocol.ILogData;
+import org.corfudb.protocols.wireprotocol.TokenResponse;
+import org.corfudb.runtime.CorfuRuntime;
+import org.corfudb.runtime.exceptions.AppendException;
+import org.corfudb.runtime.exceptions.OverwriteException;
+import org.corfudb.runtime.exceptions.StaleTokenException;
+import org.corfudb.runtime.exceptions.TrimmedException;
+import org.corfudb.runtime.view.Address;
+import org.corfudb.runtime.view.StreamOptions;
+
+
+/**
+ * This class represents a view of a single stream.
+ *
+ * <p>A SingleStreamView is tied to a stream address space, which contains an ordered set of
+ *  all the addresses (global) with log entries for this stream. A stream address space is the
+ *  entity which determines how streams are traversed. So, if a stream relies on backpointers
+ *  to traverse its address space, the SingleStreamView will use the
+ *  BackpointerStreamAddressSpace.</p>
+ *
+ *  <p>Current implementation of a stream view in Corfu, is defined with a two-fold space of
+ *  addresses, one that tracks the addresses of the regular stream and another that tracks the
+ *  addresses that have been checkpoint(ed). This composed space of addresses is represented by the
+ *  CompositeStreamAddressSpace.</p>
+ *
+ * <p>Created by amartinezman on 4/24/18.</p>
+ */
+@Slf4j
+public class SingleStreamView implements IStreamView {
+
+    final private UUID id;
+    final private CorfuRuntime runtime;
+    private IStreamAddressSpace sas;
+    final StreamOptions options;
+
+    public SingleStreamView(UUID id, CorfuRuntime runtime,
+                            BiFunction<UUID, CorfuRuntime, IStreamAddressSpace> streamAddressSpace,
```

Comment:
I feel like its more natural to have IStreamAddressSpace as the second parameter as opposed to passing a BiFunction